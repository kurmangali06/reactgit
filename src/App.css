.form {
  background-color: #15172b;
  border-radius: 20px;
  box-sizing: border-box;
  padding: 20px;
  width: 320px;
}

.input {
  display: block;
  width: 100%;
  height: calc(2.25rem + 2px);
  padding: 0.375rem 0.75rem;
  font-family: inherit;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #212529;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #bdbdbd;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.input::placeholder {
  color: #212529;
  opacity: 0.4;
}

.input:focus {
  color: #212529;
  background-color: #fff;
  border-color: #bdbdbd;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(158, 158, 158, 0.25);
}

.submit {
  display: block;
  background-color: #08d;
  border-radius: 12px;
  border: 0;
  box-sizing: border-box;
  color: #eee;
  cursor: pointer;
  font-size: 18px;
  height: 50px;
  text-align: center;
  width: 200px;
}

.list1a {
  margin: 20px 0 0;
  padding: 0;
  list-style: none;
  counter-reset: li;
  width: 200px;
}

.list1a li {
  border: 2px solid #337AB7;
  background: #EBF3FA;
  position: relative;
  margin-bottom: 20px;
  padding: 16px 10px;
}

.list1a li:hover {
  z-index: 1;
}

.list1a li:before {
  border: 2px solid #337AB7;
  position: absolute;
  top: -14px;
  padding: 2px 10px;
  font-size: 11px;
  font-weight: bold;
  color: #337AB7;
  background: #F7FAFD;
  counter-increment: li;
  content: counter(li);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.list1a li:hover:before {
  background: #337AB7;
  color: #FFF;
  -webkit-transform: translate(-12px, 0);
  -ms-transform: translate(-12px, 0);
  -o-transform: translate(-12px, 0);
  transform: translate(-12px, 0);
}

.list1a li:after {
  content: "";
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
  -webkit-transition-property: width;
  transition-property: width;
  z-index: -1;
  background: #FFF;
  height: 100%;
  left: 0;
  top: 0;
  width: 0;
}

.list1a li:hover:after {
  width: 100%;
}

.container {
  background-color: #000;
  height: 100vh;
}