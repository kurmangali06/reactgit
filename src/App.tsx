import React, { FC, useCallback, useEffect, useState } from 'react';
import { nanoid } from 'nanoid'
import { MessageList } from './Components/MessageList';
import Form from './Components/Form/Form';
import { AUTHOR } from './constants';
import style from './App.module.css'

interface Message {
  id: string
  author: string,
  value: string,
}

export const App: FC = () => {
  const [messages, setMessages] = useState<Message[]>([]);

  useEffect(() => {
    if (
      messages.length > 0 &&
      messages[messages.length - 1].author !== AUTHOR.BOT
    ) {
      const timeout = setTimeout(() => {
        setMessages([
          ...messages,
          {
            id: nanoid(),
            author: AUTHOR.BOT,
            value: `HELLO I'm BOT!! `,
          },
        ]);
      }, 1000);

      return () => {
        clearTimeout(timeout);
      };
    }
  }, [messages]);

  const addMessage = useCallback((value: string) => {
    setMessages((prevMessage) => [
      ...prevMessage,
      {
        id: nanoid(),
        author: AUTHOR.USER,
        value,
      },
    ]);
  }, []);
  return (
    <div className={style.form}>
      <MessageList messages={messages} />
      <Form addMessage={addMessage} />

    </div>
  );
};

export default App;
