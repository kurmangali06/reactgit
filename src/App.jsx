import React, { useEffect, useState } from 'react';
import './App.css';
import { MessageList } from './Components/MessageList';
import Form from './Components/Form/Form';
import { AUTHOR } from './constants';

export const App = () => {
  const [messages, setMessages] = useState([]);

  useEffect(() => {
    if (
      messages.length > 0 &&
      messages[messages.length - 1].author !== AUTHOR.BOT
    ) {
      const timeout = setTimeout(() => {
        setMessages([
          ...messages,
          {
            author: AUTHOR.BOT,
            value: "HELLO I'm BOT!! ",
          },
        ]);
      }, 1000);

      return () => {
        clearTimeout(timeout);
      };
    }
  }, [messages]);

  const addMessage = (value) => {
    setMessages([
      ...messages,
      {
        author: AUTHOR.USER,
        value,
      },
    ]);
  };
  return (
    <div className="container">
      <Form className="form" addMessage={addMessage} />
      <MessageList messages={messages} />
    </div>
  );
};

export default App;
