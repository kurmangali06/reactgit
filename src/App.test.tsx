import React from 'react';
import { App } from './App';
//import { Form } from './Components/Form/Form';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
//import { userEvent } from '@storybook/testing-library';

describe('App', () => {
  it('renders Form component', () => {
    render(<App />);
  });
  it('render with snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('is there a tag', () => {
    render(<App />);
    expect(screen.queryByText('div')).toBeNull;
  });

  /*
  не запускаеться!
  it('messages in the form', async () => {
    const moskMessage = jest.fn();
    render(<Form addMessage={() => setTimeout(moskMessage, 1000)} />);

    userEvent.click(getByText(/click/));

    await waitFor(() => expect(moskMessage).toHaveBeenCalledTimes(1), {
      timeout: 1100,
    });
  })*/
});
