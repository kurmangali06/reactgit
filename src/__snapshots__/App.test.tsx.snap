// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="form"
  >
    <ul
      class="first"
    />
    <form
      class="form"
    >
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary input css-1tcs2i1-MuiInputBase-root-MuiInput-root"
        data-testid="unput-element"
      >
        <input
          class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          placeholder="Введите текст"
          type="text"
          value=""
        />
      </div>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        click
      </button>
    </form>
  </div>
</DocumentFragment>
`;
